tox.ini

```diff
@@ -1,23 +1,38 @@
 [tox]
 envlist =
-    py{3.9,3.10}-django{32,40},
-    flake8
+    py{3.9,3.10,3.11,3.12}-django4.2
+    py{3.10,3.11,3.12,3.13}-django{5.1,5.2}
```

Comment:
Q: Do the dots matter? Or does tox-gh-actions bend the reality so it does some magic behind the scenes? Because the dj envs are defined without the dot however all is happily green…

EDIT Nah, it silently passes, but the missing env name just uses the dep without constraint, so all the individual envs end up testing against Django 5.2 as can be confirmed in the logs around freeze line.

```suggestion
    py{3.10,3.11,3.12}-django42
    py{3.10,3.11,3.12,3.13}-django{51,52}
```

Reply:
> ![Screenshot 2025-04-08 at 23 37 52](https://github.com/user-attachments/assets/e09020e8-1169-41af-ac3d-e35f594f4d9b)

Reply:
Wow, good catch there.

Reply:
Added the dots and all good.

Reply:
Yea I think in the last ~six projects I’ve encountered around ~eight different constructs how to approach this;) and almost every single one had a bug totally invisible from the bird’s eye view, only after inspecting the logs it showed that it’d run default tox instead of lint job, or just the current env instead of misspecified combo etc., mostly sneaked in via copypasta from projects with slightly different syntax, so I’m just double checking now. 🤷